src/Symfony/Component/HttpFoundation/Session/Storage/NativeSessionStorage.php

```diff
@@ -60,7 +60,7 @@ class NativeSessionStorage implements SessionStorageInterface
      * ("auto_start", is not supported as it tells PHP to start a session before
      * PHP starts to execute user-land code. Setting during runtime has no effect).
      *
-     * cache_limiter, "" (use "0" to prevent headers from being sent entirely).
+     * cache_limiter, "0"
```

Comment:
ok, maybe you have something with this PR :)
I wouldn't change this: the default should be safe when this storage is used without the SaveSessionListener.
instead, I would suggest to change this setting at the same place where SaveSessionListener is registered.

Reply:
I have a different opinion I have to admit :-) To me, Symfony‘s biggest strength has always been the abstraction of Request and Response. Saving the session doesn‘t work automatically, it‘s dependent of the SaveSessionListener. Just the same as making it private should be. I would even go further and check if the session is actually empty and if so, unset the session cookie. That, in my opinion, would be the real clean way.

Reply:
the thing you're missing is that the session subsystem doesn't know about Request/Response objects, despite being in the same component.
the thing that does the link is SaveSessionListener
which means NativeSessionStorage should have a safe default when used without Req/Resp objects (ie send "private" itself as does native php)
Then when both Session and Req/Resp are used, then SaveSessionListener does the link, and then also the cache_limiter should be set to "0".

> I would even go further and check if the session is actually empty and if so, unset the session cookie

isn't that already the case since #24523? or do you mean something else?

Reply:
I forgot to add that I think I have the same opinion as yours. I'm just trying to help finding a way to make that happen with the current code, without breaking everything :)

Reply:
I understand your point. We could merge the header onto the response object if present but we don't know if it was set by the session handler or not so that doesn't sound like a very safe idea to me. 

> isn't that already the case since #24523? or do you mean something else?

Yeah you're right, I missed that sorry :)

Reply:
Here is the patch I mean. Looks enough to me, isn't it?
```diff
diff --git a/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php b/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php
index 6f972ce..73f612d 100644
--- a/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php
+++ b/src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php
@@ -825,7 +825,7 @@ class FrameworkExtension extends Extension
 
         // session storage
         $container->setAlias('session.storage', $config['storage_id'])->setPrivate(true);
-        $options = array();
+        $options = array('cache_limiter' => '0');
         foreach (array('name', 'cookie_lifetime', 'cookie_path', 'cookie_domain', 'cookie_secure', 'cookie_httponly', 'use_cookies', 'gc_maxlifetime', 'gc_probability', 'gc_divisor', 'use_strict_mode') as $key) {
             if (isset($config[$key])) {
                 $options[$key] = $config[$key];
diff --git a/src/Symfony/Component/HttpKernel/EventListener/SaveSessionListener.php b/src/Symfony/Component/HttpKernel/EventListener/SaveSessionListener.php
index 36809b5..7a733bd 100644
--- a/src/Symfony/Component/HttpKernel/EventListener/SaveSessionListener.php
+++ b/src/Symfony/Component/HttpKernel/EventListener/SaveSessionListener.php
@@ -53,6 +53,7 @@ class SaveSessionListener implements EventSubscriberInterface
         $session = $event->getRequest()->getSession();
         if ($session && $session->isStarted()) {
             $session->save();
+            $event->getResponse()->setPrivate();
         }
     }
 ```

Reply:
Ah yes, that could work out 😄 I'm off for 2 weeks now with no internet so feel free to adjust what's needed if you feel like it cannot wait. Will work on it when I'm back if it can :) Merry X-mas @nicolas-grekas, thanks for all the hard work you put into Symfony 🎉